package.json

```diff
@@ -315,5 +315,9 @@
 		"bunyan": "^1.8.14",
 		"chokidar": "^2.1.8",
 		"eslint-nibble": "^6.0.0"
+	},
+	"volta": {
+		"node": "12.20.1",
+		"yarn": "1.22.10"
```

Comment:
How does this interact with the version set in `.yarnrc`?

Reply:
Part of the `yarn` executable's behavior is to get the correct version via `.yarnrc`, isn't it? I did a test by pinning `yarn@1.21` for volta. You can see that although volta will invoke it's yarn@1.21.1 binary, yarn reports its version is `1.22.10`. I think everything will work smoothly and ultimately `.yarnrc` determines the version like I'd expect.

```
# volta list
⚡️ Currently active tools:

    Node: v12.20.1 (current @ …/calypso/package.json)
    Yarn: v1.21.1 (current @ …/calypso/package.json)
    Tool binaries available:
        wp-env (default)
        neovim-node-host (default)
        stryker (default)
        tsc, tsserver (current @ …/calypso/package.json)
        yarn-deduplicate (default)

See options for more detailed reports by running `volta list --help`.
# yarn --version
1.22.10
```

Reply:
Seems weird and confusing to have a yarn version defined in two places. Do we really need to tell `volta` about `yarn` version?

What's the dev experience if we drop the version in `.yarnrc` and we want to update `yarn`?

Reply:
If we go all-in with volta, I believe we could safely drop either one.

> What's the dev experience if we drop the version in .yarnrc and we want to update yarn?

I don't understand the question. Updating yarn via volta means running `volta pin yarn@latest` in the repo root.

Reply:
Let's say we drop `.yarnrc` and we pin yarn using `volta` to a specific version (`1.22.10`).

How do update the version? I guess it is only a change in `package.json`, right?

What happens when a dev checks out a branch that points to a version they don't have locally? Does volta automatically downloads the new version?

Reply:
> How do update the version? I guess it is only a change in package.json, right?

Yep.

> What happens when a dev checks out a branch that points to a version they don't have locally? Does volta automatically downloads the new version?

Yes.